core/templates/pages/topics-and-skills-dashboard-page/modals/delete-skill-modal.component.spec.ts

```diff
@@ -92,4 +118,53 @@ describe('Assign Skill to Topic Modal Component', () => {
     expect(componentInstance.topicsAssignments).toEqual(testSkills);
     expect(componentInstance.topicsAssignmentsAreFetched).toBeTrue();
   });
+
+  it('should allow skill deletion', fakeAsync(() => {
+    let topicsAndSkillsDashboardBackendApiService = TestBed.inject(
+      TopicsAndSkillsDashboardBackendApiService);
+    componentInstance.skillId = 'skill_id';
+    componentInstance.topicsAssignmentsAreFetched = false;
+    spyOn(
+      topicsAndSkillsDashboardBackendApiService,
+      'fetchTopicIdToDiagnosticTestSkillIdsAsync'
+    ).and.returnValue(Promise.resolve({
+      topicIdToDiagnosticTestSkillIds: {topicId1: []}
+    }));
+    componentInstance.fetchTopicAssignmentsForSkill();
+    tick(50);
+    expect(componentInstance.skillCanBeDeleted).toBeTrue();
+  }));
+
+  it('should not allow skill deletion', fakeAsync(() => {
```

Comment:
I think this could do with more of an explanation as to why.

Reply:
Added little more explanation, please take a look.